src/amplitude-client.js

```diff
@@ -1633,6 +1688,23 @@ AmplitudeClient.prototype.sendEvents = function sendEvents() {
     checksum: md5(Constants.API_VERSION + this.options.apiKey + events + uploadTime),
   };
 
+  if (!navigator.sendBeacon) {
```

Comment:
hm, I think I would expect this in the `validateTransport` function and not in `sendEvents`